src/dist/communicator.cc

```diff
@@ -0,0 +1,143 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
move dist/* into io/ folder?

Reply:
I have moved the both communicator.cc and communicator.h to io folders

Reply:
sorry i will also need to modify the cmakelist